File: zd_01.cpp
#include <iostream>
#include <cmath>
using namespace std;

const double g = 9.81;

int main() {
    double V, T;
    cout << "Enter initial velocity (V): ";
    cin >> V;
    cout << "Enter flight time (T): ";
    cin >> T;

    double alpha = M_PI / 4;  // Start with 45 degrees (in radians)
    double tolerance = 1e-6;
    double f, df;

    do {
        f = 2 * V * sin(alpha) - g * T;   
        df = 2 * V * cos(alpha);          
        alpha -= f / df;                   
    } while (abs(f) > tolerance);         

    double alpha_deg = alpha * 180 / M_PI;

    cout << "Launch angle (radians): " << alpha << endl;
    cout << "Launch angle (degrees): " << alpha_deg << endl;

    return 0;
}

File: zd_01.py
from math import asin, degrees

def calculate_launch_angle(V, T):

    g = 9.81  
    a = asin((g * T) / (2 * V))
    return degrees(a)

V = float(input("Enter initial velocity V (m/s): "))
T = float(input("Enter flight time T (s): "))

alpha = calculate_launch_angle(V, T)
print(f"Launch angle α = {alpha:.2f}°")
